### PR TITLE
New clippies just dropped

### PR DESCRIPTION
**Pull request by itowlson**
This PR is Clippy fixes only.

References:
* https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
* https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
* https://rust-lang.github.io/rust-clippy/master/index.html#from_over_into

### CODE REVIEW

**Review by technosophos — APPROVED** [review]
Wow... pretty soon Clippy is gonna be a better Rust programmer than I am. Wait... Clippy might _already_ be a better Rust programmer than I am.

File: src/main.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, path::Path};
```
Comment (bacongobbler) [inline src/main.rs:1]: nit: should this import be part of line 2's import? IOW

```
use std::collections::HashMap;
use std::path::{Path, PathBuf};
```

**Review by bacongobbler — COMMENTED** [review]
Looks good here! Just one issue around import ordering

File: src/main.rs (hunk 1 above)
Comment (itowlson) [reply src/main.rs]: Good eye - thanks!  Fixed.

**Pull request merged**